scripts/ci/validate-metadata-files.js

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env node
+
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const CUSTOM_EXPORT_NAMES = {
+	'CKFinderUploadAdapter': 'UploadAdapter'
+};
+
+const fs = require( 'fs-extra' );
+const upath = require( 'upath' );
+const parser = require( '@babel/parser' );
+const minimist = require( 'minimist' );
+const { glob } = require( 'glob' );
+const { red, green, magenta } = require( 'chalk' );
+
+main()
+	.catch( err => {
+		console.error( err );
+
+		process.exit( 1 );
+	} );
+
+async function main() {
+	const options = getOptions( process.argv.slice( 2 ) );
+
+	console.log( magenta( '\nValidating plugins\' metadata...' ) );
+
+	const packagesGlobPattern = upath.join( options.cwd, 'packages', '*' );
+	const packagesPaths = ( await glob( packagesGlobPattern ) ).map( upath.toUnix );
+	const packagesData = await Promise.all( packagesPaths.map( getPackageData ) );
+	const packagesWithMetadata = packagesData.filter( p => p.metadata );
+
+	const invalidMetadata = [];
+
+	for ( const packageData of packagesWithMetadata ) {
+		const missingExports = getMissingExports( packageData );
+		const missingIcons = getMissingIcons( packageData, options );
+
+		if ( ![ ...missingExports, ...missingIcons ].length ) {
+			continue;
+		}
+
+		invalidMetadata.push( {
+			name: packageData.name,
+			missingExports,
+			missingIcons
+		} );
+	}
+
+	if ( !invalidMetadata.length ) {
+		console.log( green( 'Validation successful.' ) );
+
+		return;
+	}
+
+	for ( const x of invalidMetadata ) {
```

Comment:
`x`? If cannot find a proper variable name, I propose a destructing assignment. 

```suggestion
	for ( const { name, missingExports, missingIcons } of invalidMetadata ) {
```

Reply:
Oops, forgot to replace this one